src/Server/OutputParser.cs

```diff
@@ -0,0 +1,151 @@
+//-----------------------------------------------------------------------------
+// <copyright file="OutputParser.cs" company="WheelMUD Development Team">
+//   Copyright (c) WheelMUD Development Team.  See LICENSE.txt.  This file is 
+//   subject to the Microsoft Public License.  All other rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------------
+
+using WheelMUD.Utilities;
+
+namespace WheelMUD.Server
+{
+    public static class OutputParser
+    {
+        //TODO: Optimize more
+        public static string NonAnsi(char[] buffer, int bufferPos, int wordWrapLength)
+        {
+            var baseResult = "";
+            var token = "";
+            var charFromNewline = 0;
+
+            var isToken = false;
+
+            for (var i = 0; i < bufferPos; i++)
+            {
+                if (isToken && buffer[i] == '>')
+                {
+                    if (buffer[i - 1] == '%')
```

Comment:
I suspect output buffer that starts with `>` or ends with `<` will crash from indexing out of range?

Reply:
No, because of the isToken check. But I have added a check to make sure it doesn't check before or after the provider buffer just incase.